Pearson-Advance/frontend-app-lti
# feat: add table to list courses & complete deep linking flow

### Ticket

PAD-1543

### Description

This PR adds a table to list the institution-licensed courses

### Main changes

- [src/features/Courses/CoursesPage/index.jsx](https://github.com/Pearson-Advance/frontend-app-lti/pull/6/files#diff-8a3d072d2bedf7bbfadb4de6a4b6a6b4e311ed48e8f8c008b7db0fc017ebed41): Renders the table and manages the required handlers. Here, the Deep Linking flow is validated to send the selected courses to the requesting platform and is completed by submitting the HTML form obtained through the `validateContentSelection` request
- [src/features/Courses/CoursesTable/index.jsx](https://github.com/Pearson-Advance/frontend-app-lti/pull/6/files#diff-27ba03c043725a2d55a76c69267144c39c1075b61fce7c48fa2551f55db5a29d): Table itself. In future, the `actionButton` would redirect to the CCXs of the found master course ID

### Screenshots

![image](https://github.com/user-attachments/assets/657fece5-9d15-4f15-92d9-d00a44709eb2)

### How to test
- Set the environment to use `Deep Linking`
- Enable CCX in a course and create a Class of that course
- Enable the redirect to the LTI MFE when launching a `Deep Linking Flow`
- Up the LTI MFE 
- If everything goes well, once you launch a Deep Linking Flow (I use IMS), you should be redirected to the LTI MFE and you should watch the courses table with a list of the licensed master courses
